src/cachew/__init__.py

```diff
@@ -486,6 +487,68 @@ def callable_name(func: Callable) -> str:
     mod = getattr(func, '__module__', None) or ''
     return f'{mod}:{func.__qualname__}'
 
+def callable_module_name(func: Callable) -> Optional[str]:
+    return getattr(func, '__module__', None)
+
+# could cache this, but might be worth not to, so the user can change it on the fly?
+def _parse_disabled_modules(logger: Optional[logging.Logger] = None) -> List[str]:
+    # e.g. CACHEW_DISABLE=my.browser:my.reddit
+    if 'CACHEW_DISABLE' not in os.environ:
+        return []
+    disabled = os.environ['CACHEW_DISABLE']
+    if disabled.strip() == '':
+        return []
+    if ',' in disabled and logger:
+        logger.warning('CACHEW_DISABLE contains a comma, but this expects a $PATH-like, colon-separated list; '
+                       f'try something like CACHEW_DISABLE={disabled.replace(",", ":")}')
+    return disabled.split(':')
+
+
+def _matches_disabled_module(module_name: str, pattern: str) -> bool:
+    '''
+    >>> _matches_disabled_module('my.browser', 'my.browser')
+    True
+    >>> _matches_disabled_module('my.browser', 'my.*')
+    True
+    >>> _matches_disabled_module('my.browser', 'my')
+    True
+    >>> _matches_disabled_module('my.browser', 'my.browse*')
+    True
+    >>> _matches_disabled_module('my.browser.export', 'my.browser')
+    True
+    >>> _matches_disabled_module('mysomething.else', '*')  # CACHEW_DISABLE='*' disables everything
+    True
+    >>> _matches_disabled_module('my.browser', 'my.br?????')  # fnmatch supports unix-like patterns
+    True
+    >>> _matches_disabled_module('my.browser', 'my.browse')
+    False
+    >>> _matches_disabled_module('mysomething.else', 'my')  # since not at '.' boundary, doesn't match
+    False
+    >>> _matches_disabled_module('mysomething.else', '')
+    False
+    '''
+    import fnmatch
+
+    if module_name == pattern:
+        return True
+
+    for mp, pp in zip(module_name.split('.'), pattern.split('.')):
```

Comment:
hmm, wouldn't that disable the module if the pattern is longer than the module name? 
i.e. `_matches_disabled_module('my.browser', 'my.browser.export')` is `True`?

Reply:
Ah right, zip just uses the smaller length... will compare the lengths beforehand and if the `len(module_parts) < len(pattern_parts)` will just early exit